fvm/environment/derived_data_invalidator.go

```diff
@@ -22,10 +22,47 @@ type ContractUpdate struct {
 type DerivedDataInvalidator struct {
 	ContractUpdateKeys []ContractUpdateKey
 	FrozenAccounts     []common.Address
+
+	MeterParamOverridesUpdated bool
 }
 
 var _ derived.TransactionInvalidator = DerivedDataInvalidator{}
 
+func NewDerivedDataInvalidator(
+	contractKeys []ContractUpdateKey,
+	env *facadeEnvironment,
+) DerivedDataInvalidator {
+	return DerivedDataInvalidator{
+		ContractUpdateKeys:         contractKeys,
+		FrozenAccounts:             env.FrozenAccounts(),
+		MeterParamOverridesUpdated: meterParamOverridesUpdated(env),
+	}
+}
+
+func meterParamOverridesUpdated(env *facadeEnvironment) bool {
+	updatedRegisterIds, _ := env.txnState.RegisterUpdates()
+
+	serviceAccount := string(env.chain.ServiceAddress().Bytes())
+	storageDomain := common.PathDomainStorage.Identifier()
+	for _, registerId := range updatedRegisterIds {
+		if registerId.Owner != serviceAccount {
+			continue
+		}
```

Comment:
Can we add some comments to explain why we skip the non service account? Just in case the assumption changes in the future.
```suggestion
		if registerId.Owner != serviceAccount {
			continue
		}
```

Reply:
will do.